ksql-engine/src/main/java/io/confluent/ksql/util/ArrayUtil.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright 2017 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.ksql.util;
+
+import java.lang.reflect.Array;
+
+public class ArrayUtil {
+
+  public static <T> int getNullIndex(T[] array) {
+    for (int i = 0; i < array.length; i++) {
+      if (array[i] == null) {
+        return i;
+      }
+    }
+    return -1;
+  }
+
+  public static <T> T[] getNoNullArray(Class<T> clazz, T[] array) {
+    int nullIndex = getNullIndex(array);
+    if (nullIndex == -1) {
+      return array;
+    }
+    T[] noNullArray = (T[]) Array.newInstance(clazz, nullIndex);
+    for (int i = 0; i < noNullArray.length; i++) {
+      noNullArray[i] = array[i];
+    }
+    return noNullArray;
+  }
+
+  public static <T> T[] padWithNull(Class<T> clazz, T[] array, int finalLength) {
+    if (array.length >= finalLength) {
+      return array;
+    }
+    T[] paddedArray = (T[]) Array.newInstance(clazz, finalLength);
+    for(int i = 0; i < array.length; i++) {
+      paddedArray[i] = array[i];
+    }
+    for (int i = array.length; i < finalLength; i++) {
+      paddedArray[i] = null;
+    }
+    return paddedArray;
+  }
+
+  public static <T> boolean containsValue(T value, T[] array) {
+    for (T v: array) {
+      if (v != null && v.toString().equals(value.toString())) {
```

Comment:
what's the need to convert valuet to string? Why not directly compare value with `equals`?

Reply:
Good point! No need for toString. Removed it.

Reply:
@hjafarpour while working on JSON_ARRAY_CONTAINS udf, I found a use case for converting to the string. When the parser converts json number as a long and search value is int conversion to string is needed. Example:

```
Object valueLong = (Object) 1L;
Object valueInt = (Object) 1;
valueLong.equals(valueInt); //false
```

Reply:
thought, if compare with toString() then it leads to unsound comparison logic.

```
Object valueString = (Object) "1";
Object valueInt = (Object) 1;
valueString.equals(valueInt); //false
valueString.toString().equals(valueInt.toString()); //true
```